internal/versions/versions.go

```diff
@@ -105,12 +105,15 @@ func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params o
 		}
 		if swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
 			if !checkedForArmAuthorization {
-				err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture)
-				if err != nil {
-					return common.GenerateErrorResponder(err)
+				checkedForArmAuthorization = true
+				if err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture); err != nil {
+					if strings.Contains(err.Error(), auth.InvalidCPUArchitecture) {
+						continue
+					} else {
+						return common.GenerateErrorResponder(err)
```

Comment:
can be handled later, but shouldn't checkedForArmAuthorization flag be true in this case? this error could be just some temporary communication issue with ams, so worth try again. or, maybe add caching to the response instead of the flag?

Reply:
Thanks for the feedback. We discussed this over slack.
The question is: shouldn't `checkedForArmAuthorization` flag be false in this case?

The answer is: no
The reason: The idea behind the bool `checkedForArmAuthorization` is for the `V2ListSupportedOpenshiftVersions` function to contact AMS one time per function call. The `V2ListSupportedOpenshiftVersions` function may iterate over multiple images with an `arm64` CPU arch. If we get an error from AMS (not a lack of capability), we will stop and return at line 113.